docs/http/http-spans.md

```diff
@@ -382,6 +403,26 @@ Following attributes MUST be provided **at span creation time** (when provided a
 
 Note that in some cases host and port identifiers in the `Host` header might be different from the `server.address` and `server.port`, in this case instrumentation MAY populate `Host` header on `http.request.header.host` attribute even if it's not enabled by user.
 
+## Failed requests
+
+HTTP request may end prematurely if it was cancelled or an error occurred preventing
+the client or server from reading/writing the request/response.
+
+On the server side it means that at least one of the following conditions has happened:
+- request headers were not received or response headers were not sent in full
+- attempt to read request body or write response body failed
+
+On the client side it means:
+- request headers were not sent or response headers were not received in full
+- attempt to write request body failed
+- attempt to read response body failed in the scope of instrumented portion of HTTP request flow.
+
+Spans for requests that end with an error before response headers are sent/received do not have `http.response.status_code`
+attribute set and SHOULD have `error.reason` attribute describing a class of such error.
+
+If an error happened in the scope of request instrumentation, but after headers are sent/received, corresponding spans SHOULD have
+both `http.response.status_code` and `error.reason` attributes set.
```

Comment:
s/error.reason/error.description/
```suggestion
Spans for requests that end with an error before response headers are sent/received do not have `http.response.status_code`
attribute set and SHOULD have `error.description` attribute describing a class of such error.

If an error happened in the scope of request instrumentation, but after headers are sent/received, corresponding spans SHOULD have
both `http.response.status_code` and `error.description` attributes set.
```

Reply:
thanks, great catch! After implementing @trask suggestion to use `error.description` whenever a request fails (with exception or 500), this section seems redundant